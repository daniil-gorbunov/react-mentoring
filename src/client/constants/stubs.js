export const testMovie = {
  id: 1,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: 'backdrop.jpg',
  poster_path: 'poster.jpg',
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieWoPosters = {
  id: 2,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: null,
  poster_path: null,
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieList = [
  testMovie,
  testMovieWoPosters,
];
