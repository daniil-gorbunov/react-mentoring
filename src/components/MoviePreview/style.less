@import '../../assets/styles/colors';

.movie-preview {
  color: @text-color-semi-dark;
  min-height: 460px;
  margin-top: 40px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .poster {
    height: 100%;
  }

  .text-info {
    margin: 0 60px;

    .main-info {
      .title {
        text-transform: uppercase;
        font-weight: 600;
      }

      .year {
        font-size: 0.7em;
        border: 1px solid @text-color-semi-dark;
        border-radius: 2px;
        font-weight: bold;
        line-height: 18px;
        padding: 0 5px;
      }
    }

    .genre {
      font-size: 0.7em;
    }
  }
}
