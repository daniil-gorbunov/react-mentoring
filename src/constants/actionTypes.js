export const CHANGE_SEARCH_QUERY = 'CHANGE_SEARCH_QUERY';
export const CHANGE_SEARCH_TYPE = 'CHANGE_SEARCH_TYPE';
export const CHANGE_SORT_TYPE = 'CHANGE_SORT_TYPE';
export const SORT_MOVIES = 'SORT_MOVIES';
export const SEARCH_MOVIES = 'SEARCH_MOVIES';
