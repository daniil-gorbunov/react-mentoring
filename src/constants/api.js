export const CONFIG = {
  IMAGES: {
    BASE_URL: 'http://image.tmdb.org/t/p/',
    POSTER_SIZES: {
      XS: 'w92',
      S: 'w154',
      M: 'w185',
      L: 'w342',
      XL: 'w500',
      XXL: 'w780',
      ORIGINAL: 'original',
    },
  },
};
export const BASE_URL = 'https://api.themoviedb.org/3/';
export const V3_KEY = '74613ce7f9c68dc71edc73f9216b3dd5';
