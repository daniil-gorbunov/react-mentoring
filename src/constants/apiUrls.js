export default 'https://netflixroulette.net/api/api.php';
