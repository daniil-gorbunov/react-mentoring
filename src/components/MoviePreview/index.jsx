import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Row, Col } from 'react-flexbox-grid';
import { DIRECTOR } from '../../constants/searchTypes';
import { openMovie, searchMovies } from '../../actions';
import commonStyles from '../../assets/styles/common.less';
import styles from './style.less';
import movieType from '../../types/movieType';

const Movie = ({ movie, onMovieClick }) => (
  <div className={styles.moviePreview}>
    <Link
      to={`/film/${encodeURIComponent(movie.show_title)}`}
      onClick={() => onMovieClick(movie)}
    >
      <div>
        <Row>
          <Col xs={12}>
            <Row center="xs">
              <Col xs={12}>
                <img className={styles.poster} src={movie.poster} alt={movie.show_title} />
              </Col>
            </Row>
          </Col>
        </Row>
        <div className={styles.textInfo}>
          <Row>
            <Col xs={12}>
              <div className={styles.mainInfo}>
                <Row between="xs">
                  <Col xs={10}>
                    <span className={styles.title}>{movie.show_title}</span>
                  </Col>
                  <Col xs={2}>
                    <div className={commonStyles.textRight}>
                      <span className={styles.year}>{movie.release_year}</span>
                    </div>
                  </Col>
                </Row>
              </div>
              <Row>
                <Col xs={12}>
                  <span className={styles.genre}>{movie.category}</span>
                </Col>
              </Row>
            </Col>
          </Row>
        </div>
      </div>
    </Link>
  </div>
);

Movie.propTypes = {
  movie: movieType,
  onMovieClick: PropTypes.func.isRequired,
};

Movie.defaultProps = {
  movie: null,
};

const mapStateToProps = (state, ownProps) => ({
  movie: ownProps.movie || state.movie,
});

const mapDispatchToProps = dispatch => ({
  onMovieClick: (movie) => {
    dispatch(openMovie(movie));
    searchMovies(dispatch, movie.director, DIRECTOR);
  },
});

const MoviePreview = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Movie);

export default MoviePreview;
