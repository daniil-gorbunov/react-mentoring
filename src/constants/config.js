import { CONFIG } from './api';

export const IMG_URL = CONFIG.IMAGES.BASE_URL;
export const POSTER_SIZE = CONFIG.IMAGES.POSTER_SIZES.L;
export const NO_IMAGE_PATH = '/src/assets/images/no-image.png';
