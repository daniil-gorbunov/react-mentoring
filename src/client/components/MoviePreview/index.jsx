import moment from 'moment';
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Row, Col } from 'react-flexbox-grid';
import commonStyles from '../../assets/styles/common.less';
import styles from './style.less';
import movieType from '../../types/movieType';
import { IMG_URL, POSTER_SIZE, NO_IMAGE_PATH } from '../../constants/config';

const MoviePreview = ({ movie, onMovieClick }) => (
  <div className={styles.moviePreview}>
    <Link
      to={`/film/${encodeURIComponent(movie.title)}`}
      onClick={() => onMovieClick(movie)}
    >
      <div>
        <Row>
          <Col xs={12}>
            <Row center="xs">
              <Col xs={12}>
                <img
                  className={styles.poster}
                  src={movie.poster_path ? `${IMG_URL}${POSTER_SIZE}${movie.poster_path}` : NO_IMAGE_PATH}
                  alt={movie.original_title}
                />
              </Col>
            </Row>
          </Col>
        </Row>
        <div className={styles.textInfo}>
          <Row>
            <Col xs={12}>
              <div className={styles.mainInfo}>
                <Row between="xs">
                  <Col xs={10}>
                    <span className={styles.title}>{movie.original_title}</span>
                  </Col>
                  <Col xs={2}>
                    <div className={commonStyles.textRight}>
                      <span className={styles.year}>{moment(movie.release_date).format('YYYY')}</span>
                    </div>
                  </Col>
                </Row>
              </div>
            </Col>
          </Row>
        </div>
      </div>
    </Link>
  </div>
);

MoviePreview.propTypes = {
  movie: movieType,
  onMovieClick: PropTypes.func.isRequired,
};

MoviePreview.defaultProps = {
  movie: null,
};

export default MoviePreview;
