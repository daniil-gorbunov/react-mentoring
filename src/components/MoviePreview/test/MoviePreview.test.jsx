import React from 'react';
import renderer from 'react-test-renderer';
import { testMovie, testMovieWoPosters } from '../../../constants/stubs';
import MoviePreview from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
const clickHandler = jest.fn();

describe('MoviePreview', () => {
  test('renders with poster', () => {
    const preview = renderer.create(<MoviePreview
      movie={testMovie}
      onMovieClick={clickHandler}
    />);
    expect(preview).toMatchSnapshot();
  });

  test('renders without poster', () => {
    const preview = renderer.create(<MoviePreview
      movie={testMovieWoPosters}
      onMovieClick={clickHandler}
    />);
    expect(preview).toMatchSnapshot();
  });
});
