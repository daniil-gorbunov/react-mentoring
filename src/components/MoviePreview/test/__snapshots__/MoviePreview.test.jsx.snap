// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreview renders with poster 1`] = `
<div
  className="moviePreview"
>
  <Link
    onClick={[Function]}
    replace={false}
    to="/film/Test%20Movie"
  >
    <div>
      <Row>
        <Col
          xs={12}
        >
          <Row
            center="xs"
          >
            <Col
              xs={12}
            >
              <img
                alt="Test Movie"
                className="poster"
                src="http://image.tmdb.org/t/p/w342poster.jpg"
              />
            </Col>
          </Row>
        </Col>
      </Row>
      <div
        className="textInfo"
      >
        <Row>
          <Col
            xs={12}
          >
            <div
              className="mainInfo"
            >
              <Row
                between="xs"
              >
                <Col
                  xs={10}
                >
                  <span
                    className="title"
                  >
                    Test Movie
                  </span>
                </Col>
                <Col
                  xs={2}
                >
                  <div
                    className="textRight"
                  >
                    <span
                      className="year"
                    >
                      2010
                    </span>
                  </div>
                </Col>
              </Row>
            </div>
          </Col>
        </Row>
      </div>
    </div>
  </Link>
</div>
`;

exports[`MoviePreview renders without poster 1`] = `
<div
  className="moviePreview"
>
  <Link
    onClick={[Function]}
    replace={false}
    to="/film/Test%20Movie"
  >
    <div>
      <Row>
        <Col
          xs={12}
        >
          <Row
            center="xs"
          >
            <Col
              xs={12}
            >
              <img
                alt="Test Movie"
                className="poster"
                src="/src/assets/images/no-image.png"
              />
            </Col>
          </Row>
        </Col>
      </Row>
      <div
        className="textInfo"
      >
        <Row>
          <Col
            xs={12}
          >
            <div
              className="mainInfo"
            >
              <Row
                between="xs"
              >
                <Col
                  xs={10}
                >
                  <span
                    className="title"
                  >
                    Test Movie
                  </span>
                </Col>
                <Col
                  xs={2}
                >
                  <div
                    className="textRight"
                  >
                    <span
                      className="year"
                    >
                      2010
                    </span>
                  </div>
                </Col>
              </Row>
            </div>
          </Col>
        </Row>
      </div>
    </div>
  </Link>
</div>
`;
